File: TAbaretata/JavaScript-Fundamentals-2015/08.Strings/06.ExtractTextFromHTML/ExtractTextFromHTML.js
/*
Problem 6: Write a function that extracts the content of a html page given as text. 
The function should return anything that is in a tag, without the tags:
    <html><head><title>Sample site</title></head><body><div>text<div>more text</div>and more...</div>in body</body></html>
Result:
    Sample sitetextmore textand more...in body
*/

function extractTextFromHTML(text) {
    var answer = '',
        inTag = false,
		i;
		
    for (i = 0; i < text.length; i+=1) {
        if (text[i] === '<') {
            inTag = true;
        } else if (text[i] === '>') {
            inTag = false;
        } else if (!inTag) {
            answer += text[i];
        }
    }
    return answer;
}

var text = '<html><head><title>Sample site</title></head><body><div>text<div>more text</div>and more...</div>in body</body></html>';
console.log(extractTextFromHTML(text));

File: TAbaretata/CSharpPartTwo-2015/StringsAndTextProcessing/25.ExtractTextFromHTML/ExtractTextFromHTML.cs
using System;
using System.Linq;
using System.Text.RegularExpressions;

/*Write a program that extracts from given HTML file its title (if available), and its body text without the HTML tags.
Example input:                                                              Output:
<html>
  <head><title>News</title></head>                                          Title: News
  <body><p><a href="http://academy.telerik.com">Telerik                     Text: Telerik Academy aims to provide  
    Academy</a>aims to provide free real-world practical                    free real-world practical training
    training for young people who want to turn into                         for young people who want to turn 
    skilful .NET software engineers.</p></body>                             into skilful .NET software engineers.   
</html>             */

class ExtractTextFromHTML
{
    static void Main()
    {
        string text = "<html><head><title>News</title></head><body><p><a href=\"http://academy.telerik.com\">Telerik Academy</a>aims to provide free real-world practical training for young people who want to turn into skilful .NET software engineers.</p></body></html>";
        
        MatchCollection tags = Regex.Matches(text, @"((?<=^|>)[^><]+?(?=<|$))");
        int count = 1;

        foreach (Match tag in tags)
        {
            if (count == 1)
            {
                Console.WriteLine("Title: {0}", tag);
                Console.Write("Text: ");
            }
            else
            {
                Console.Write(tag + " ");
            }
            count++;
        }
        Console.WriteLine();
    }
}
